### PR TITLE
refactored nutrition calculator to typescript

### PR DESCRIPTION
**Pull request by thinklikeadesigner**
In this PR, I 
- refactored the TypeScript Nutrition Calculator class into 4 utility functions, 
- removed the JS file
- updated /bmr to exclude BMR type as it's no longer needed due to BF being required
- renamed calculateMacronutrients to getNutritionInfo, which now returns 
```js 
{ protein, fat, carb, calories, bmr, tdee }
```

`getNutritionInfo` is the sole export of NutritionCalculations.ts since, it holds all the answers

### CODE REVIEW

File: src/utils/NutritionCalculations.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+			['extraActive', 1.9]
+		]);
+		const level = activityFactors.get(activityLevel);
+		if (level) {
+			const tdee = bmr * level;
+			return tdee;
+		}
+	}
+	console.log('input invalid need bmr and/or activity level');
+	return false;
```
Comment (mentalcaries) [inline src/utils/NutritionCalculations.ts:43]: For functions with a return type of `number` , I've been returning `0` in the event of an error or invalid value. 
That way you don't have to use a union operator

File: src/utils/NutritionCalculations.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
+const calculateBMR = (weight: number, height: number, age: number, sex: string): number => {
+	const weightInKg = weight / 2.20462;
+	const heightInCm = height * 2.54;
+	let bmr;
+	if (sex === 'male') {
+		bmr = 10 * weightInKg + 6.25 * heightInCm - 5 * age + 5;
+	} else {
+		bmr = 10 * 65.77 + 6.25 * 163 - 5 * 29 - 161;
```
Comment (mentalcaries) [inline src/utils/NutritionCalculations.ts]: This returns the same value for all females. I'm not sure what the original formula is.

File: src/utils/NutritionCalculations.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+} => {
+	const bmr = calculateBMRWithBF1(weight, height, age, sex, bf);
+	const tdee = Number(calculateTDEE(bmr, activityLevel));
+
+	const protein = Math.round(weight);
+
+	const fat = Math.round((tdee * 0.25) / 9);
+	const carb = Math.round((tdee - protein * 4 - fat * 9) / 4);
+	const calories = protein * 4 + carb * 4 + fat * 9;
+	return { protein, fat, carb, calories };
```
Comment (mentalcaries) [inline src/utils/NutritionCalculations.ts:68]: Fantastic!! Makes me want to consume macros right now

File: src/utils/NutritionCalculations.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+console.log(calculateMacronutrients());
+console.log(calculateBMR(145, 64, 29, 'female'));
+console.log(calculateBMRWithBF1(145, 64, 29, 'female', 0.22));
+console.log(calculateTDEE(calculateBMR(145, 64, 29, 'female'), 'sedentary'));
+
```
Comment (mentalcaries) [inline src/utils/NutritionCalculations.ts:71]: If we're leaving these in for testing, can we create an issue to clean them up when we're done?

File: src/utils/NutritionCalculations.ts (hunk 1 above)
Comment (thinklikeadesigner) [reply src/utils/NutritionCalculations.ts:43]: fixed it

File: src/utils/NutritionCalculations.ts (hunk 4 above)
Comment (thinklikeadesigner) [reply src/utils/NutritionCalculations.ts:71]: deleted these

**Review by mentalcaries — APPROVED** [review]
Nice work!! 🚀

**Pull request merged**